src/DependencyInjection/AmqpExtension.php

```diff
@@ -134,12 +134,16 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('publishers.xml');
         $loader->load('services.xml');
 
-        if ('php_extension' === $config['driver']) {
-            $loader->load('driver/php-extension.xml');
-        }
-
-        if ('php_lib' === $config['driver']) {
-            $loader->load('driver/php-lib.xml');
+        switch ($config['driver']) {
+            case 'php_extension':
+                $loader->load('driver/php-extension.xml');
+                break;
+            case 'php_lib':
+                $loader->load('driver/php-lib.xml');
+                break;
+            case 'php_lib_sockets':
+                $loader->load('driver/php-lib-sockets.xml');
+                break;
```

Comment:
By default you must throw error, because you not know what can be.

Reply:
It just reproduces the previous behaviour. There was no "else" condition